editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx

```diff
@@ -491,17 +491,49 @@ export const AnalyticsProvider: React.FC<{ children: React.ReactNode }> = ({
   }
 
   function getAllowListAnswers(
+    nodeId: string,
     breadcrumb: Store.userData,
-  ): Record<string, any>[] | undefined {
+  ): Record<string, unknown>[] | undefined {
+    const answers = getAnswers(nodeId);
+    const data = getData(breadcrumb);
+
+    const allowListAnswers = [...answers, ...data];
+    if (!allowListAnswers.length) return;
+
+    return allowListAnswers;
+  }
+
+  /**
+   * Extract allowlist answers from user answers
+   * e.g. from Checklist or Question components
+   */
+  function getAnswers(nodeId: string) {
```

Comment:
I think `getAnswers` isn't combining all of the values set by a key:


<img width="629" alt="Screenshot 2024-03-01 at 10 35 05" src="https://github.com/theopensystemslab/planx-new/assets/36415632/d0d3d0cc-79bb-4cc2-8abe-dd75dfea5433">